Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py

```diff
@@ -267,6 +268,43 @@ def populate(self, issue, member=None):
 
         return issue
 
+    def set(self, issue, assignee=None, opened=None, why=None, **properties):
+        update_dict = dict()
+
+        if assignee:
+            if not isinstance(assignee, User):
+                raise TypeError("Must assign to '{}', not '{}'".format(User, type(assignee)))
+            issue._assignee = self.user(name=assignee.name, username=assignee.username, email=assignee.email)
+            update_dict['assignees'] = [issue._assignee.username]
+
+        if opened is not None:
+            issue._opened = bool(opened)
+            update_dict['state'] = 'open' if issue._opened else 'closed'
+
+        if properties:
+            raise TypeError("'{}' is an invalid property".format(list(properties.keys())[0]))
+        if update_dict:
+            update_dict['number'] = [issue.id]
+            response = requests.patch(
+                '{api_url}/repos/{owner}/{name}/issues/{id}'.format(
+                    api_url=self.api_url,
+                    owner=self.owner,
+                    name=self.name,
+                    id=issue.id,
+                ), auth=HTTPBasicAuth(*self.credentials(required=True)),
```

Comment:
Not decline this change, but should we consider use OAuth instead? Since GitHub recommend OAuth

Reply:
Would be interested in having a larger discussion about this. I think setting up OAuth for command line tools will be a bit rough. We'd probably need to have folks register a code generator and a password instead of a token (which is what they do now). This also wouldn't work for instances authenticated with something else, like `SAML`, so we'd probably need to retain the API token workflow.